blocks/blogsearch/blogsearch.js

```diff
@@ -1,15 +1,69 @@
 import {
-  div, form, input, i,
+  a, div, form, input, i,
 } from '../../scripts/dom-helpers.js';
+import { getLocaleInfo, debounce } from '../../scripts/scripts.js';
+import ffetch from '../../scripts/ffetch.js';
+
+async function handleSearch(block) {
+  const searchValue = block.querySelector('input').value;
+  if (searchValue.length < 3) {
+    return;
+  }
+
+  const searchResults = block.querySelector('.search-results');
+  searchResults.innerHTML = '';
+
+  console.log(`searching for ${searchValue}`);
+
+  const entries = await ffetch(`${getLocaleInfo().metadataIndex}`)
+    .sheet('blogs-content')
+    .filter((entry) => {
```

Comment:
Before doing a search based on term, we should filter out any blog pages which are not part of the current locale.  This can be done before storing the data in memory.

Reply:
so we're not building an index per locale? As far as I understand there's no use case for a cross-locale search, so indexes restricted to locale should benefit performance, right?

Reply:
Yes, it would, but it comes at the cost of duplicating this config https://github.com/hlxsites/servicenow/blob/main/helix-query.yaml and all the xcel formulas for every locale. If any would change in the future, it would need to be updated in all places. So for the moment that's why we went with having on big index in `/blogs/query-index.json` and doing a little bit of client side filtering. For all the other usescase, except full text search, the difference in content size is not that much.

What we could have as a compromise solution, if you think it is better, just for the `blogs-content` because it has more information, we could have one xcel sheet per locale, like `blogs-content-en-US`, `blogs-content-fr-FR`, etc. I'm not sure how much performance it would bring. probably more for the Geosites since the US has the majority of content. I would leave this up to you to decide. We can start with 1 sheet, have the final implementation and if it feels slow, do a split.

Reply:
in 7d07f47